ocaml/lang/best-practice/list.ml

```diff
@@ -0,0 +1,13 @@
+let test1 xs =
+  (* ruleid:list-find-outside-try *)
+  if List.find 1 xs
+  then 1
+  else 2
+
+let test2 xs =
+ (* ok *)
+ try 
+   if List.find 1 xs
+   then 1
+   else 2
+ with Not_found -> 3
```

Comment:
I would also recommend using `List.find_opt`.

Similarly, there's `Hashtbl.find`, which used to be widely used and could use a semgrep rule (if you haven't added one already). There must be a few others similar functions that raise `Not_found` in `List` or in other modules of the stdlib.

Note that Janestreet Base/Core_kernel/Core clobbers the interface of the standard library. Users of that library normally do `open Base` or `open Core_kernel` or `open Core`, all of which expose a [different `List.find`](https://ocaml.janestreet.com/ocaml-core/latest/doc/base/Base/List/index.html#val-find). I wouldn't worry about it.